### ISSUE TITLE
Please add more documentation

### ISSUE THREAD

**Issue by 0HD** [Soaku/DiscordGD#1]
It's still unclear to me how to use your module, even though I read it a bunch of times, for almost 5 months now (from time to time of course). I keep remembering this module and get back to it, but still not know how to implement it into my project.

> ...you need to connect with Discord using `start(app_id: String)`.
The issue lies here. I have no idea where to use `start` in the first place. I've tried it everywhere, but still won't work. Also, where do I set the properties for the module?

**Comment by Soaku** [comment]
You should first load the [gdns script](https://github.com/Soaku/DiscordGD/blob/master/gdnative/libdiscord.gdns) located in the repo, then instance it and call the method.

```gdscript
const Discord = preload("res://gdnative/libdiscord.gdns")

var discord = Discord.new()

func _ready():
    discord.start("your app ID")
```

**Comment by P7i** [comment]
Thank you so much for the quick response!
I have yet to try that, but I really appreciate your help anyway!

**Comment by 0HD** [comment]
Thank you so much for the quick response!
I have yet to try that, but I really appreciate your help anyway.

**Comment by 0HD** [comment]
Hi, I've copied your code and pasted it into the root node's script, but sadly it didn't work.

![image](https://user-images.githubusercontent.com/64617192/82591896-b44d5780-9ba8-11ea-8a97-68b3b2534104.png)

It also shows 2 errors:

> **E 0:00:00.660   _get_modified_time: Failed to get modified time for: D:/.../bin/libdiscord.gdnlib.**
  <C++ Error>   Method failed. Returning: 0
  <C++ Source>  drivers/windows/file_access_windows.cpp:344 @ _get_modified_time()
**E 0:00:00.662   init_library:  does not have a library for the current platform.**
  <C++ Error>   Condition "lib_path.length() == 0" is true.
  <C++ Source>  modules/gdnative/nativescript/nativescript.cpp:1483 @ init_library()

FYI I'm using Godot 3.2.1.stable on Windows 10.0.18362.

**Comment by Soaku** [comment]
Are you sure you compiled the library? Looks like it couldn't access the lib's `dll` file.

**Comment by simonboi** [comment]
It feels difficult for me too. I also get errors. I also don't know how to compile, even when trying to read the docs. I would like some help with instructions..

**Comment by Soaku** [comment]
If you have trouble compiling, you can always get precompiled binaries on the [release page](https://github.com/Soaku/DiscordGD/releases). It would be nice to share what errors do you get.

**Comment by 0HD** [comment]
I've been trying to compile it as well, but compiling it requires the C++ module in Visual Studio 2017 (and I don't have that module, nor do I want to reinstall VS).
I was also about to say it's difficult, but you said I don't have to compile it. I'm already using the release version, though. What could be the issue?

**Comment by Soaku** [comment]
Sorry for the lack of activity lately. Generally, if you download DiscordGD.zip and unpack it to your project directory, it should be available to use immediately. If you unpack it to a subdirectory, then you'll have to update the paths used in gdnative/libdiscord.gdnlib and gdnative/libdiscord.gdns so they point to the correct files.

**Comment by craftablescience** [comment]
I figured out there's an error in the binaries - the `libdiscord.gdns` file points to `res://bin/libdiscord.gdnlib` for the library instead of `res://gdnative/libdiscord.gdnlib`. Hope this helps anyone else encountering the "null instance" error.

**Comment by ViniciosLugli** [comment]
Even correcting the path error, it executes the code and does nothing, the RPC is not modified :(, Any solution?

**Comment by Soaku** [comment]
@ViniciosLugli are you calling `update()`?

**Comment by craftablescience** [comment]
I have a way around it sometimes not updating immediately in the `ready()` function. First I set the image to display on Discord in the `_ready` function. Next, I set up an autoload timer for 1 second, and update the Discord status at this point with the relevant text info. It always shows the right status now. Thank you so much @Soaku for making this library!

**Comment by ViniciosLugli** [comment]
now that i remembered that i am using godot 3.2, it must be this .. i did everything right and followed the timer tip but he doesn’t show an answer, i will try with a godot 3.1.

**Comment by Soaku** [comment]
@ViniciosLugli I don't think that can be the issue. Can you show your code?

**Comment by ViniciosLugli** [comment]
![image](https://user-images.githubusercontent.com/40807526/96013887-28b8b980-0e1c-11eb-8ac4-066eab13655e.png)
the timer is 1 second

**Comment by Soaku** [comment]
@ViniciosLugli Assuming you provide a valid API token to `start()` I don't see anything wrong here. Do you get any errors?

**Comment by craftablescience** [comment]
![image](https://user-images.githubusercontent.com/26600014/96302292-930d5d80-0fc6-11eb-8583-c8d0be1cbf3d.png)
Try setting the discord variables here too.

**Comment by ZRCoding** [comment]
This won't work for me at all. I spent hours trying to get it to work, on multiple different Godot versions. Even with the help of @craftablescience in the Godot Discord, we weren't able to find a solution. Even after fixing the path error in libdiscord.gdns, I am still getting a null instance error.

We went as far as him sending me a working project using this lib, and when I ran it it didn't work. I am totally lost and defeated, hopefully you can help out. Thank you

**Comment by Soaku** [comment]
@ZRCoding …sounds like an issue with Discord itself. The lib requires the Discord desktop app, and the game must run alongside it.

**Comment by craftablescience** [comment]
It's definitely not that, there should still be an instance loaded after `preload("res://gdnative/libdiscord.gdns")`, regardless of whether Discord is running or not. Just tried running my app without Discord running and no null instance error.

Honestly his computer is cursed lol

**Comment by Soaku** [comment]
You should all probably use this library instead of my own https://github.com/WesleiRamos/Godot-Rich-Presence

**Comment by craftablescience** [comment]
That one isn't being maintained either from the looks of it. Also, GameSDK has no GDScript bindings as of now. I wonder if I could fix that...

**Comment by Soaku** [comment]
Yes, it isn't being maintained, but it at least seems to be done better than my library, is more complete than my library, and isn't built full of bugs.

Offers no pre-built binaries however.
